js/jquery.terminal-1.3.1.js

```diff
@@ -3759,12 +3759,12 @@
         // :: validating if object is a string or a function, call that function
         // :: and display the exeption if any
         // ---------------------------------------------------------------------
-        function validate(label, object) {
+        function validate(label, object, term) {
             try {
                 if ($.isFunction(object)) {
                     object(function() {
                         // don't care
-                    });
+                    }, term);
```

Comment:
It would be better to just use self here, not need to defined term parameter:
```
object($.noop, self);
```
for `this` to work with prompt (I've missed this one when I added `this` to all callbacks) each instance of prompt like in `set_prompt` (maybe that's the only place, not sure right now) command will need to have `prompt.call(self, callback, self);` also validate will need to have `object.call(self, $.noop, self);`